client/jetpack-connect/plans.jsx

```diff
@@ -38,65 +37,77 @@ import {
 	canCurrentUser,
 	getJetpackConnectRedirectAfterAuth,
 	isRtl,
+	isSiteAutomatedTransfer,
 	isSiteOnPaidPlan,
 } from 'state/selectors';
 import {
 	getFlowType,
-	isRedirectingToWpAdmin,
 	getSiteSelectedPlan,
 	getGlobalSelectedPlan,
 	isCalypsoStartedConnection,
 } from 'state/jetpack-connect/selectors';
 import { mc } from 'lib/analytics';
-import { isSiteAutomatedTransfer } from 'state/selectors';
+import { isJetpackSite } from 'state/sites/selectors';
 
 const CALYPSO_REDIRECTION_PAGE = '/posts/';
 const CALYPSO_PLANS_PAGE = '/plans/my-plan/';
 const JETPACK_ADMIN_PATH = '/wp-admin/admin.php?page=jetpack';
 
 class Plans extends Component {
-	static propTypes = { showJetpackFreePlan: PropTypes.bool };
-
 	static defaultProps = { siteSlug: '*' };
 
-	redirecting = false;
-
 	componentDidMount() {
-		if ( this.props.isAutomatedTransfer && ! this.redirecting && this.props.selectedSite ) {
-			this.redirecting = true;
-			this.props.goBackToWpAdmin( this.props.selectedSite.URL + JETPACK_ADMIN_PATH );
-		} else if ( this.hasPreSelectedPlan() ) {
-			this.autoselectPlan();
-		} else {
+		if ( ! this.maybeRedirect( this.props ) ) {
 			this.props.recordTracksEvent( 'calypso_jpc_plans_view', {
 				user: this.props.userId,
```

Comment:
Let's just move this to the controller and get rid of the `maybeRedirect` return as you suggest.

Reply:
@johnHackworth Just want to check, would it be a problem for the funnels if we fire `calypso_jpc_plans_view` every time we hit the /plans route, even if we don't render the component?

It would let us simplify the code a bit, and currently the attempts to not fire it depend on what state happens to be loaded.

Reply:
I don't think it wouldn't be any problem for the funnels, but we would lose the raw number of people "landing" on plans page. Maybe we could add a new different event to count that in the controller?

Reply:
There are a few events in the controller that might work:

https://github.com/Automattic/wp-calypso/blob/b4e0a3e8f300436351786255393cc6703ca8a603/client/jetpack-connect/controller.js#L200-L208

@johnHackworth Would that work, or do we need something completely new?

Reply:
Proposal is to move this event to the controller, so we should capture that information, but it may be inflated because it will count users that pass through the plans step without seeing the component.

Happy to leave this as is for now, and keep an eye on the numbers.